### PR TITLE
Feature Optional dependencies documentation updates

### PR DESCRIPTION
**Pull request by mauicv**
## What is this

This work branches off [`feature/dep-management`](https://github.com/SeldonIO/alibi-detect/pull/501) and Incorporates changes to the documentation for the dependency management functionality.

# Notes:

- The main issue here is how to handle optional dependencies for `conda` installs. We've come to the conclusion that we should keep the `conda` install large. This is opposed to not including certain dependencies and then documenting what extra dependencies are required to enable certain functionalities. The main reason for this decision was that we're kind of held hostage by the rest of the conda recipe community who seem to opt for larger installs themselves. For instance, it's impossible for us to make PyTorch optional without making transformers optional because PyTorch is a dependency of the transformers conda recipe. We could take the approach of trying to figure out what dependencies we need to eliminate in order to get the optional dependencies we want to enable but this will be very hard to maintain and test for. For instance, if another dependency adds or removes a dependency we want to keep optional then we have to be alerted to it to update our dependencies.
- Because PyTorch is included in transformers I've also removed that from the conda install tab in the install docs!

___

### What this PR doesn't include:

1. Changes to the notebooks resulting from changes to the public API are included in the PRs (`tf, tf-tfp` and `torch`) where those changes are implemented, not here.

### CODE REVIEW

File: doc/source/overview/getting_started.md [hunk 1]
```diff
@@ -9,54 +9,186 @@ by following the instructions below.
...
 
 ```bash
 pip install alibi-detect
 ```
+
+```{div} sd-mb-1
+The default installation provides out the box support for the following detectors:
+- ChiSquareDrift
+- CVMDrift
+- FetDrift
```
Comment (ascillitoe) [inline doc/source/overview/getting_started.md:40]: `FetDrift` -> `FETDrift`

File: doc/source/overview/getting_started.md [hunk 2]
```diff
@@ -9,54 +9,186 @@ by following the instructions below.
...
+- LearnedKernelDrift
+- LSDDDrift
+- LSDDDriftOnline
+- MMDDrift
+- MMDDriftOnline
+- SpotTheDiffDrift
+- ContextMMDDrfit
+
+The TensorFlow installation is required to use the following detectors
+- OutlinerAE
```
Comment (ascillitoe) [inline doc/source/overview/getting_started.md:147]: `OutlinerAE` -> `OutlierAE`

File: doc/source/overview/getting_started.md [hunk 3]
```diff
@@ -9,54 +9,186 @@ by following the instructions below.
...
-Installation with [TensorFlow](https://www.tensorflow.org/) and [PyTorch](https://pytorch.org/) backends.
+Installation with [PyTorch](https://pytorch.org/) backend.
 ```
 
 ```bash
 pip install alibi-detect[torch]
 ```
 
+```{div} sd-mb-1
+The PyTorch installation is required to use the PyTorch backend for the following detectors
```
Comment (ascillitoe) [inline doc/source/overview/getting_started.md:106]: ":" at end of line

File: README.md [hunk 4]
```diff
@@ -181,8 +185,8 @@ The following tables show the advised use cases for each algorithm. The column *
 
 #### TensorFlow and PyTorch support
 
-The drift detectors support TensorFlow and PyTorch backends. Alibi Detect does however not install PyTorch for you. 
-Check the [PyTorch docs](https://pytorch.org/) how to do this. Example:
+The drift detectors support TensorFlow and PyTorch backends. Alibi Detect does not install these as default. See the 
+[installation options](#installation-and-usage) for more details
```
Comment (ascillitoe) [inline README.md:189]: "." at end of line.

**Review by ascillitoe — COMMENTED** [review]
Mostly looks good to me. A few typo's need correcting.

File: doc/source/overview/getting_started.md [hunk 5]
```diff
@@ -76,36 +208,9 @@ conda install mamba -n base -c conda-forge
...
-````
-
-````{tab-item} PyTorch
-:sync: label-torch
-:class-label: sd-pt-0
-
-Installation with [TensorFlow](https://www.tensorflow.org/) and [PyTorch](https://pytorch.org/) backends.
-
-```bash
-mamba install -c conda-forge alibi-detect pytorch
```
[the remark this answers is not in the document]
Comment (mauicv) [reply doc/source/overview/getting_started.md:99]: Nope, I'll write something in the PR intro.

**Comment by ascillitoe** [comment]
We might also want to remove the `'OutlierProphet',  # requires prophet: pip install alibi-detect[prophet]` line later on in `getting_started.md`. Or else add similar comments to the other detectors that require optional installs.

**Review by ascillitoe — APPROVED** [review]
LGTM, subject to the duplicate `conftest` issue with `mypy` being resolved!
